src/sinks/databend/api.rs

```diff
@@ -0,0 +1,213 @@
+use std::collections::BTreeMap;
+
+use bytes::Bytes;
+use http::Request;
+use http::StatusCode;
+use hyper::Body;
+use serde::{Deserialize, Serialize};
+
+use crate::{http::Auth, http::HttpClient, sinks::util::UriSerde};
+
+use super::error::DatabendError;
+
+#[derive(Serialize, Debug)]
+pub struct StageAttachment {
+    location: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    file_format_options: Option<BTreeMap<String, String>>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    copy_options: Option<BTreeMap<String, String>>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct DatabendHttpRequest {
+    sql: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    stage_attachment: Option<StageAttachment>,
+}
+
+impl DatabendHttpRequest {
+    pub const fn new(sql: String) -> Self {
+        Self {
+            sql,
+            stage_attachment: None,
+        }
+    }
+
+    pub fn add_stage_attachment(
+        &mut self,
+        location: String,
+        file_format_options: Option<BTreeMap<String, String>>,
+        copy_options: Option<BTreeMap<String, String>>,
+    ) {
+        self.stage_attachment = Some(StageAttachment {
+            location,
+            file_format_options,
+            copy_options,
+        });
+    }
+}
+
+#[derive(Deserialize, Debug)]
+pub struct DatabendHttpResponseSchemaField {
+    pub name: String,
+    pub r#type: String,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct DatabendHttpResponseError {
+    pub code: i64,
```

Comment:
Does this need to be signed, could it just be `usize` or `u32`?

Reply:
changed to `u16`